From 6479296e31892868de2e980da2b3c639fdfa7ffd Mon Sep 17 00:00:00 2001
From: Daniel Rettenmaier <danielrettenmaier@gmail.com>
Date: Thu, 14 Mar 2019 21:40:41 +0100
Subject: [PATCH] BugFix: dont flip surfaceVectorFields

---
 .../mapping/fvFieldMappers/MapFvSurfaceField.H       | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/finiteVolume/interpolation/mapping/fvFieldMappers/MapFvSurfaceField.H b/src/finiteVolume/interpolation/mapping/fvFieldMappers/MapFvSurfaceField.H
index 40dfd16f7..7550bb806 100644
--- a/src/finiteVolume/interpolation/mapping/fvFieldMappers/MapFvSurfaceField.H
+++ b/src/finiteVolume/interpolation/mapping/fvFieldMappers/MapFvSurfaceField.H
@@ -71,6 +71,18 @@ void MapInternalField<Type, MeshMapper, surfaceMesh>::operator()
 
     field.autoMap(mapper.surfaceMap());
 
+////////////////////////////////////////////////////////////////////////
+// Andera Montorfano, Politecnico de Milano. Thank you very much!
+// Daniel Deising fixed the idea :)
+
+    // MUST not change sign of vector fields!!!
+//    if (field.typeName == vectorField::null().typeName)       // bug by Andrea
+    if (std::is_same<Type, vector>::value) //vector::typeName)  // fixed by Daniel Deising
+    {
+       return;
+    } 
+////////////////////////////////////////////////////////////////////////
+
     // Flip the flux
     const labelList flipFaces = mapper.surfaceMap().flipFaceFlux().toc();
 
-- 
2.21.0

